### PR TITLE
SFA circular harmonics

### PR DESCRIPTION
**Pull request by narahahn**
* Circular array on the horizontal plane with a uniform distribution
* Radial and angular functions for plane waves and line sources (z-independent only)
* Modal analysis for plane wave decomposition

### CODE REVIEW

**Comment by ghost** [comment]
## CHT ordering

Currently the circular harmonics are ordered from `-N,...,0,...N`.
I think `0,...,N,-N,...,-1` would be better for consistency with the regular DFT.


## CHT scaling

IMHO the CHT should be normalized such that `FWF^H = I`, where `F` is a DFT matrix and `W` is a diagonal matrix with constant diagonal `w`.

Three possibilities:

1. `w = 1` and put `1/sqrt(M)` into `F`

2. `w = 1/M`

3. `w = 2pi/M` and put `1/sqrt(2pi)` into `F`

My personal preference would be 2.

File: micarray/modal/angular.py [hunk 1]
```diff
@@ -91,6 +91,48 @@ def Legendre_matrix(N, ctheta):
...
+    if pol.ndim == 0:
+        Q = 1
+    else:
+        Q = len(pol)
+    if weights is None:
+        weights = np.ones(Q)
+    Psi = np.zeros([(2*N+1), Q], dtype=complex)
+    order = np.arange(-N, N+1)
+    for i, n in enumerate(order):
+        Psi[i, :] = np.exp(1j * n * pol)
```
Comment (ghost) [inline micarray/modal/angular.py]: Weights are not applied!

File: micarray/modal/angular.py [hunk 2]
```diff
@@ -153,3 +195,23 @@ def grid_gauss(n):
     weights = np.repeat(weights, 2*n+2)
     weights *= np.pi / (n+1)      # sum(weights) == 4pi
     return azi, elev, weights
+
+
+def grid_equal_polar_angle(M, phi0=0):
```
Comment (ghost) [inline micarray/modal/angular.py]: The spherical grid functions take the maximum `N` as the only argument.
For consistency I'd prefer the same behavior. (i.e. pass `N` and get `2N+1` positions.)

**Review by ghost — CHANGES_REQUESTED** [review]

File: micarray/modal/angular.py (hunk 2 above)
Comment (ghost) [reply micarray/modal/angular.py]: @narahahn prefers the current version, being more useful in practice.
I prefer the less flexible but more consistent interface.

@spors What's your opinion?

**Comment by narahahn** [comment]
* `grid_equal_polar_angle` now returns the weights of `1/M`, with `M` being the number of microphones
* The `weights` is applied to the CHT matrix in `cht_matrix`
* The order of CHT orders is changed: 0,...,N,-N,...,-1 (`cht_matrix`, `mirror_vec`)

File: micarray/modal/angular.py [hunk 3]
```diff
@@ -91,6 +91,48 @@ def Legendre_matrix(N, ctheta):
...
+        Maximum order.
+    pol : (Q,) array_like
+        Polar angle.
+    weights : (Q,) array_like, optional
+        Weights.
+
+    Returns
+    -------
+    Psi : (2N+1, Q) numpy.ndarray
+        Matrix of spherical harmonics.
```
Comment (ghost) [inline micarray/modal/angular.py]: spherical -> circular

File: micarray/modal/angular.py [hunk 4]
```diff
@@ -91,6 +91,48 @@ def Legendre_matrix(N, ctheta):
...
 
 
+def cht_matrix(N, pol, weights=None):
+    r"""Matrix of circular harmonics up to order N for given angles.
+
+    Computes a matrix of circular harmonics up to order :math:`N`
+    for the given angles/grid.
+
+    .. math::
+        \Psi = \left[ \begin{array}{ccccccc}
```
Comment (ghost) [inline micarray/modal/angular.py:101]: The formatting is off.

**Review by ghost — APPROVED** [review]

**Pull request merged**